### ISSUE TITLE
Provide target option to navigateTo

### ISSUE THREAD

**Issue by Mokkapps** [nuxt/nuxt#20625]
### Describe the feature

We have a UI component library that should be completely decoupled from Nuxt. Therefore, we want to inject a callback function that is called if a anchor tag is clicked and which calls `navigateTo`. 

But `navigateTo` provides no possibility to add target options like `_blank` to open the to path in a new tab.

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by wilsonpinto** [comment]
While still under discussion, I decided to proactively open a [PR](https://github.com/nuxt/nuxt/pull/21333) as I have found a good way to start exploring the Nuxt core codebase. I fully comprehend if the team decides not to proceed with this approach.

Considerations for my changes:

- Using the [open](https://developer.mozilla.org/en-US/docs/Web/API/Window/open) method programmatically with the `_blank` target is not generally recommended, as most browsers might block it, perceiving it as a popup. This could potentially encourage bad web practices.

- The concept of a _navigateTo_ router helper with an `openNewTab` option may seem counterintuitive, as opening a new tab isn't really in line with the concept of continuing the current router navigation.

If there's an alternate approach that you believe would be more effective, I would be glad to continue my efforts in that direction.

Thanks : )
